### PR TITLE
feat: add etcd elector for gocron

### PR DESCRIPTION
**Pull request by rfyiamcool**
### summary

add etcd elector for gocron.

because the etcd test library was not found, the etcd service needs to be started before unit testing.

**refer**

https://github.com/go-co-op/gocron/pull/561

### CODE REVIEW

**Comment by JohnRoesler** [comment]
you could pull in the github repo from go cron so the tests will run 😄 https://github.com/go-co-op/gocron-redis-lock/tree/main/.github

File: elector.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	}()
+
+	for e.ctx.Err() == nil {
+		select {
+		case resp := <-electionHandler.Observe(e.ctx):
+			if len(resp.Kvs) == 0 {
+				continue
+			}
+
+			val := string(resp.Kvs[0].Value)
```
[the remark this answers is not in the document]
Comment (rfyiamcool) [reply elector.go]: done.

After my test, I received only one kv every time. But for the sake of code robustness, `for range` is still used.

File: elector.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+func (e *Elector) unsetLeader(id string) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	e.isLeader = false
+	e.leaderID = id
+}
+
+func (e *Elector) Start(electionPath string) error {
```
[the remark this answers is not in the document]
Comment (rfyiamcool) [reply elector.go:145]: done. 😃

**Comment by rfyiamcool** [comment]
> you could pull in the github repo from go cron so the tests will run 😄 https://github.com/go-co-op/gocron-redis-lock/tree/main/.github

done.

I'm really sorry that I just noticed that this PR has been updated. I always thought you didn't see this PR. 

哈哈 I'm sorry, this is my fault.

**Comment by JohnRoesler** [comment]
no worries! Thanks for adding this 😄

**Pull request merged**